e2e/node/acp-v3.test.ts

```diff
@@ -0,0 +1,271 @@
+  UrlString,
+  acp_v3 as acp,
+  FetchError,
+} from "../../src/index";
+import { getTestingEnvironment, TestingEnvironment } from "./util/getTestingEnvironment";
+import { supportsAcp } from "./util/supportsAcp";
+import { getSession } from "./util/getSession";
+
+let env: TestingEnvironment;
+
+beforeAll(() => {
+  config({
+    path: __dirname,
+    // Disable warning messages in CI
+    silent: process.env.CI === "true",
+  });
+  env = getTestingEnvironment();
+});
+
+describe(`Authenticated end-to-end ACP V3 tests against environment ${env.environment}`, () => {
+  if (!supportsAcp(env.pod)) {
+    return;
+  }
+
+  let options: any;
+  let session: Session;
+  let sessionResource: string;
+
+  beforeEach(async () => {
+    session = await getSession(env);
+    sessionResource = `${env.pod}solid-client-tests/node/acp-v3-test-dataset-${session.info.sessionId}`;
+    options = { fetch: session.fetch };
+    await saveSolidDatasetAt(sessionResource, createSolidDataset(), options);
+  });
+
+  afterEach(async () => {
+    await deleteSolidDataset(sessionResource, options);
+    await session.logout();
+  });
+
+  async function initialisePolicyResource(
+    policyResourceUrl: UrlString,
+    session: Session
+  ) {
+    let publicRule = acp.createRule(policyResourceUrl + "#rule-public");
+    publicRule = acp.setPublic(publicRule);
+
+    let publicReadPolicy = acp.createPolicy(
+      policyResourceUrl + "#policy-publicRead"
+    );
+    // Note: we should think of a better name for "optional", as this isn't really optional.
+    //       At least one "optional" rule should apply, and since this is the only rule for this
+    //       policy, it will in practice be required.
+    publicReadPolicy = acp.addAnyOfRuleUrl(publicReadPolicy, publicRule);
+    publicReadPolicy = acp.setAllowModes(publicReadPolicy, {
+      read: true,
+      append: false,
+      write: false,
+    });
+
+    let selfRule = acp.createRule(policyResourceUrl + "#rule-self");
+    selfRule = acp.addAgent(selfRule, session.info.webId!);
+    // This policy denies write access to the current user,
+    // but allows write access so the Resource can still be removed afterwards:
+    let selfWriteNoReadPolicy = acp.createPolicy(
+      policyResourceUrl + "#policy-selfWriteNoRead"
+    );
+    selfWriteNoReadPolicy = acp.addAllOfRuleUrl(
+      selfWriteNoReadPolicy,
+      selfRule
+    );
+    selfWriteNoReadPolicy = acp.setAllowModes(selfWriteNoReadPolicy, {
+      read: false,
+      append: true,
+      write: true,
+    });
+    selfWriteNoReadPolicy = acp.setDenyModes(selfWriteNoReadPolicy, {
+      read: true,
+      append: false,
+      write: false,
+    });
+
+    let policyResource = createSolidDataset();
+    policyResource = setThing(policyResource, publicRule);
+    policyResource = setThing(policyResource, publicReadPolicy);
+    policyResource = setThing(policyResource, selfRule);
+    policyResource = setThing(policyResource, selfWriteNoReadPolicy);
```

Comment:
Should we encourage the pattern
```
const policyResource = [publicRule, publicReadPolicy, selfRule, selfWriteNoReadPolicy].reduce(setThing, createSolidDataset());
```
? It's something I have seen before, and it allows declaring the `policyResource` as a `const`, but I don't know readability-wise how it compares to having multiple assignments.

We could even consider adding a `setThingAll(dataset, things)` function to the API if this becomes recurrent enough.

Reply:
I am not against it, but this is not a re-write, just a separation of ACP V3 tests which will eventually by phased out, I'll make sure I keep it in mind when adding new tests or refactoring the existing ones.